### PR TITLE
fix href setter and no_scheme edge case

### PR DESCRIPTION
**Pull request by anonrig**
Fixes https://github.com/nodejs/node/issues/46755

### CODE REVIEW

**Comment by lemire** [comment]
Merging.

**Comment by TimothyGu** [comment]
@anonrig Could you contribute the extra tests (`ada_extra_urltestdata.json`) to WPT at some point? I believe they'd be useful for other implementations as well.

File: src/parser.cpp [hunk 1]
```diff
@@ -126,7 +126,7 @@ namespace ada::parser {
         case ada::state::NO_SCHEME: {
           ada_log("NO_SCHEME ", helpers::substring(url_data, input_position));
           // If base is null, or base has an opaque path and c is not U+0023 (#), validation error, return failure.
-          if (base_url == nullptr || (base_url->has_opaque_path && (input_position != input_size))) {
+          if (base_url == nullptr || (base_url->has_opaque_path && (url_data[input_position] != '#'))) {
```
[the remark this answers is not in the document]
Comment (pthier) [reply src/parser.cpp]: This is indeed unsafe and causes OOB reads. E.g. for empty url or `#x`. This should probably be a check against `url.fragment.has_value()`?

[the remark this answers is not in the document]
Comment (lemire) [reply src/parser.cpp]: @pthier Please see https://github.com/ada-url/ada/pull/285

I wrote that it was potentially unsafe, but I later examined the code. I believe it is safe.

**Pull request merged**